nens_auth_client/tests/test_invitation_view.py

```diff
@@ -73,6 +96,25 @@ def test_invitation_not_acceptable(rf, get_object_or_404, invitation):
     assert not invitation.accept.called
 
 
+def test_invitation_email_mismatch(
+    rf, get_object_or_404, invitation, invited_user
+):
+    """Non-acceptable invitations give 404, also for anonymous users"""
```

Comment:
Een 404 is misschien niet altijd even duidelijk, maar aan de andere kant wil je ook niet teveel hints aan aanvallers geven met gedetailleerde info.

Aandachtspunt: 404 wordt vaak niet meegenomen door sentry => nog even checken of er wel ergens een duidelijke logger.raargh() staat.

Reply:
Ik heb hier nog even naar gekeken. Het bleek dat de docstring niet goed was. Als er een email-mismatch is, dan krijgt de gebruiker een 403.

Als je dit checkt in het geval je anonymous bent, dan krijg je altijd een 302 naar het login scherm.